app.js

```diff
@@ -421,6 +442,29 @@ app.post('/uploadimage', function (req, res) {
         }
     });
 });
+//get gitlab parameters
+app.get('/gitlab', function (req, res) {
```

Comment:
Here, I don't think use a root path `/gitlab` is a great idea.
Don't easily use root level path, because it will consume other path usage.
Since gitlab have a auth path, maybe we should move this action to there,